packages/use-contractkit/src/connectors/connectors.ts

```diff
@@ -330,7 +329,10 @@ export class WalletConnectConnector implements Connector {
       return undefined;
     }
     const accounts = await this.kit.contracts.getAccounts();
-    return accounts.getWalletAddress(address);
+    const walletAddress = await accounts.getWalletAddress(address);
+    return walletAddress === '0x0000000000000000000000000000000000000000'
+      ? address
+      : walletAddress;
```

Comment:
Checking if the `BigNumber.from(walletAddress).isZero()` would be more robust here.